peewee_async/pool.py

```diff
@@ -68,14 +78,74 @@ async def create(self) -> None:
         """
         if "connect_timeout" in self.connect_params:
             self.connect_params['timeout'] = self.connect_params.pop("connect_timeout")
-        self.pool = cast(
-            PoolProtocol,
-            await aiopg.create_pool(
-                database=self.database,
-                **self.connect_params
-            )
+        self.pool = await aiopg.create_pool(
+            database=self.database,
+            **self.connect_params
+        )
+
+
+class PsycopgPoolBackend(PoolBackend):
+    """Asynchronous database connection pool based on psycopg + psycopg_pool libraries.
+    """
+
+    async def create(self) -> None:
+        """Create connection pool asynchronously.
+        """
+        params = self.connect_params.copy()
+        pool = psycopg_pool.AsyncConnectionPool(
+            format_dsn(
+                'postgresql',
+                host=params.pop('host'),
+                port=params.pop('port'),
+                user=params.pop('user'),
+                password=params.pop('password'),
+                path=self.database,
+            ),
+            kwargs={
+                'cursor_factory': psycopg.AsyncClientCursor,
+                'autocommit': True,
+            },
+            **params,
         )
 
+        await pool.open()
+        self.pool = pool
+
+    def has_acquired_connections(self) -> bool:
```

Comment:
I don't know how it works but if i run single with has_acquired_connections check for example **test_transaction_success** i get an error. If i run all tests once in a time all is fine.